RELEASE.md

```diff
@@ -172,6 +172,89 @@ Steps:
 
 ---
 
+## 🍎 LLVM Apple Containers Release Process
+
+> **📋 Status**: Infrequent releases (approximately every 3 years)
+> **Use Case**: Cross-compilation for macOS targets from Linux environments
+
+### When to Release LLVM Containers
+- New Xcode/macOS SDK versions are available
+- Security updates for LLVM toolchain
+- Compatibility requirements for new macOS targets
+- Critical bug fixes in the cross-compilation toolchain
+
+### Prerequisites
+- Access to Buildkite pipeline: https://buildkite.com/elastic/llvm-apple
+- Understanding of LLVM/Clang toolchain versions
+- Knowledge of macOS SDK compatibility requirements
+
+### Step-by-Step Instructions
+
+#### 1. Prepare for Release
+- Verify the current LLVM/Clang version in the containers
+- Check for any pending updates or security patches
+- Review macOS SDK compatibility requirements
+
+#### 2. Trigger Release Build
+1. Navigate to the Buildkite pipeline: https://buildkite.com/elastic/llvm-apple
+2. Click "New Build" button
+3. Select branch: `main`
+4. Set environment variable: `RELEASE=true`
+5. Add build message: "Release LLVM Apple containers - [reason/version]"
+6. Click "Create Build"
+
+#### 3. Monitor Build Process
+- Monitor the build progress in Buildkite console
+- Expected build time: 4-5 hours (due to LLVM compilation)
+- Watch for any compilation or packaging errors
+- Verify all target architectures are built successfully
+
+#### 4. Verify Release
+After successful build completion:
+```bash
+# Check available container images
+docker pull docker.elastic.co/beats-dev/golang-crossbuild:llvm-apple-debian11-arm64
+docker inspect docker.elastic.co/beats-dev/golang-crossbuild:llvm-apple-debian11-arm64 | grep org.label-schema.build-date
+```
+
+Expected image tags:
+- `docker.elastic.co/beats-dev/golang-crossbuild:llvm-apple-debian<version>(-arm64)?`
+
+#### 5. Test Cross-Compilation
+Verify the containers work correctly.
+
+#### 6. Update Documentation
+- Update any version references in README.md
+- Document any breaking changes or new features
+- Update compatibility matrix if applicable
+
```

Comment:
Not sure where the best place in the docs to mention this would be, but it might be worth noting what you said in the PR description:

> Notify that mixing changes in the LLVM and in the go folders are not a good practice.

Expanding on that a bit: that the Go images should be built after the LLVM images.

Reply:
Do you mind contributing to this PR and including those relevant notes?